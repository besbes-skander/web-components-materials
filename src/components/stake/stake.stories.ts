import './stake'

export default {
    title: 'Components/Stake',
}

export const Default = () => '<arl-stake></arl-stake>'