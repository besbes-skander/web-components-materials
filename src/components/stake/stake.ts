import { UPDATE_STAKE } from '../../shared'

const template = document.createElement('template')
template.innerHTML = `
<div class="stake">
    <h3 class="stake__title">Votre Mise</h3>
    <div class="stake__input">
        <p>Choisir votre mise</p>
        <input type="number" value="" />
    </div>
</div>
`

export class Stake extends HTMLElement {
    constructor() {
        super()

        this.attachShadow({ mode: 'open' })
            .appendChild(template.content.cloneNode(true))

        this.shadowRoot?.querySelector('input')?.addEventListener('keyup', this.emitStake.bind(this))
    }

    emitStake() {
        const stake = this.shadowRoot?.querySelector('input')?.value
        window.dispatchEvent(new CustomEvent(UPDATE_STAKE, { detail: { stake } }))
    }
}

customElements.define('arl-stake', Stake)