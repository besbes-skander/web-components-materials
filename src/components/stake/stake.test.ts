import { render, dispatchMockedEventWith } from '../../utils'
import { Stake } from './stake'

let dispatchEvent: (key: string, value: string | number) => void
let stake: HTMLElement

const spyDispatchEvent = jest.spyOn(window, 'dispatchEvent')

describe('Stake Component', () => {
    beforeEach(async () => {
        stake = await render(Stake)
        dispatchEvent = dispatchMockedEventWith(stake)
    })

    it('should render correctly the component', () => {
        expect(stake?.shadowRoot?.innerHTML).toMatchSnapshot()
    })

    it('should emit new stake event when user choose a stake', () => {
        // Given
        const userInput = '2'

        // When
        dispatchEvent('keyup', userInput)

        // Then
        const expectedStake = (spyDispatchEvent.mock.calls[0][0] as CustomEvent).detail.stake
        expect(expectedStake).toBe(userInput)
    })
})