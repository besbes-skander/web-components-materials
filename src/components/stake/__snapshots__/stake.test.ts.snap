// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stake Component should render correctly the component 1`] = `
"
<div class=\\"stake\\">
    <h3 class=\\"stake__title\\">Votre Mise</h3>
    <div class=\\"stake__input\\">
        <p>Choisir votre mise</p>
        <input type=\\"number\\" value=\\"\\">
    </div>
</div>
"
`;
